src/d3a/models/powerflow/__init__.py

```diff
@@ -0,0 +1,123 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from abc import ABC, abstractmethod
+
+from d3a.models.strategy.load_hours import LoadHoursStrategy
+from d3a.models.strategy.storage import StorageStrategy
+from d3a.models.strategy.pv import PVStrategy
+from d3a.models.strategy.finite_power_plant import FinitePowerPlant
+from d3a.models.strategy.infinite_bus import InfiniteBusStrategy
+
+
+class PowerFlowBase(ABC):
+    def __init__(self, root_area, root_voltage):
+        """
+        :param root_area: contains tree structured hierarchical energy grid
+        :param root_voltage: contains the voltage of the top most hierarchy
+        """
+        self.root_voltage = root_voltage
+        self._grid_forming(root_area)
+
+    def _grid_forming(self, area):
+        """
+        :param area: takes area object and convert it to specific energy device
+        :return: electrical network compliant connected grid
+        """
+        if area.strategy is None:
+            area.bus = self.create_bus(area, self.root_voltage)
+            if area.parent is not None:
+                area.line = self.add_line(area.parent, area)
+        elif isinstance(area.strategy, InfiniteBusStrategy):
+            area.ext_grid = self.add_external_grid(area)
+        elif isinstance(area.strategy, LoadHoursStrategy):
+            area.load_device = self.add_load_device(area, area.strategy.avg_power_W)
+        elif isinstance(area.strategy, StorageStrategy):
+            area.ess_device = self.add_storage_device(area)
+        elif isinstance(area.strategy, PVStrategy):
+            area.pv_device = \
+                self.add_generation_device(area, area.strategy.max_panel_power_W * 1e-3)
+        elif isinstance(area.strategy, FinitePowerPlant):
```

Comment:
Great job, the only thing that I am missing here is the commercial producer. It can also be added here by creating a generation device with infinite energy (sys.maxsize).